src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1265,40 +1250,60 @@ private void Save([NotNull]Document doc, C4Document** outDoc, C4Document* baseDo
+        private void SaveResolvedDocument([CanBeNull]Document resolvedDoc, [NotNull]Document localDoc, [NotNull]Document remoteDoc)
         {
-            if (!ReferenceEquals(resolved, localDoc)) {
-                resolved.Database = this;
+            if (resolvedDoc == null) {
+                if (localDoc.IsDeleted)
+                    resolvedDoc = localDoc;
+
+                if (remoteDoc.IsDeleted)
+                    resolvedDoc = remoteDoc;
+            }
+
+
+            if (resolvedDoc!=null&&!ReferenceEquals(resolvedDoc, localDoc)) {
+                resolvedDoc.Database = this;
             }
 
-            // The remote branch has to win, so that the doc revision history matches the server's
+            // The remote branch has to win, so that the doc revision history matches the server's.
             var winningRevID = remoteDoc.RevID;
             var losingRevID = localDoc.RevID;
 
             FLSliceResult mergedBody = (FLSliceResult)FLSlice.Null;
-            if (!ReferenceEquals(resolved, remoteDoc)) {
-                // Unless the remote revision is being used as-is, we need a new revision:
-                try {
-                    mergedBody = resolved.Encode();
-                } catch (ObjectDisposedException) {
-                    WriteLog.To.Sync.E(Tag, "Resolved document for {0} somehow got disposed!",
-                        new SecureLogString(resolved.Id, LogMessageSensitivity.PotentiallyInsecure));
-                    throw new RuntimeException(
-                        "Resolved document was disposed before conflict resolution completed.  Please file a bug report at https://github.com/couchbase/couchbase-lite-net");
+            C4RevisionFlags mergedFlags = 0;
+            if (!ReferenceEquals(resolvedDoc, remoteDoc)) {
+                var isDeleted = true;
+                if (resolvedDoc != null) {
+                    // Unless the remote revision is being used as-is, we need a new revision:
+                    try {
+                        mergedBody = resolvedDoc.Encode();
+                    } catch (ObjectDisposedException) {
+                        WriteLog.To.Sync.E(Tag, "Resolved document for {0} somehow got disposed!",
+                            new SecureLogString(resolvedDoc.Id, LogMessageSensitivity.PotentiallyInsecure));
+                        throw new RuntimeException(
+                            "Resolved document was disposed before conflict resolution completed.  Please file a bug report at https://github.com/couchbase/couchbase-lite-net");
+                    }
+                    if (mergedBody.Equals((FLSliceResult)FLSlice.Null))
+                        throw new RuntimeException("Resolved document contains a null body");
+                    isDeleted = resolvedDoc.IsDeleted;
+                } else {
+                    mergedBody = (FLSliceResult)FLSlice.Allocate("");
```

Comment:
Is there a reason to use `""`?  If it has to be used instead of `null` then this makes me nervous.  Mixing allocations from C# with frees from LiteCore is not a good idea.

Reply:
Since mergedBody initialized as FLSlice.Null, I guess I can remove the logic ```else {
                    mergedBody = (FLSliceResult)FLSlice.Allocate("");
                }``` ?

Reply:
After tested with setting mergedBody to FLSlice.Null, ConflictResolver returns null test failed.

Reply:
Fails how?

Reply:
this happens, when the local copy is deleted and resolvedDoc will be null and if we try to save a null body through `c4doc_resolveConflict `, it will end up [here](https://github.com/couchbase/couchbase-lite-core/blob/ecf16ecb29ac139e7af8e7d411ac8631344cdd91/LiteCore/Database/TreeDocument.cc#L304) without entering inside it, to save the new revision.